packages/actor-rdf-parse-html-script/lib/ActorRdfParseHtmlScript.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {ActionContext, Actor, Mediator} from "@comunica/core";
+import {Readable} from "stream";
+
+/**
+ * A HTML RDF Parse actor that listens on the 'rdf-parse' bus.
+ *
+ * It is able to extract JSON-LD from HTML files and parse the JSON-LD based RDF serializations
+ * and announce the presence of them by media type.
+ */
+export class ActorRdfParseHtmlScript extends ActorRdfParseFixedMediaTypes {
+
+  public mediatorRdfParse: Mediator<Actor<IActionRootRdfParse, IActorTestRootRdfParse,
+    IActorOutputRootRdfParse>, IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+    Promise<IActorRdfParseOutput> {
+
+    const myQuads = new Readable({objectMode: true});
+
+    let initialized = false;
+
+    myQuads._read = async () => {
+      if (!initialized) {
+        initialized = true;
+
+        // Stringify HTML input
+        const htmlString: string = await require('stream-to-string')(action.input);
+
+        // Extract script tags
+        const DOMParser = require('xmldom').DOMParser;
+        const doc = new DOMParser().parseFromString(htmlString, 'text/html');
+        const scripts = doc.getElementsByTagName('script');
+
+        const supportedTypes = [
+          "application/ld+json",
+          "application/n-quads",
+          "application/n-triples",
+          "application/trig",
+          "text/n3",
+          "text/turtle",
+        ];
+
+        // Loop script elements
+        let count: number = 0;      // count quad streams
+        for (let i = 0; i < scripts.length; i++) {
```

Comment:
You can probably simplify this loop with a [for-of loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)

Reply:
Type Error: scripts is not iterable